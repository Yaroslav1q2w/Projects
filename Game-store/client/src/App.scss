* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

body {
    background-color: rgb(25,26,28);
    font-family: 'Montserrat', sans-serif;
}

.wrapper {
    //max-width: 1100px;
    margin: 0 auto;
    color: white;
    
    .main__content {
        margin-top: 14px;
    }
    
    a {
        text-decoration: none;
        color: #FFFFFF;
    }
}