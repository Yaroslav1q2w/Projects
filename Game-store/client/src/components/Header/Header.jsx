import { useEffect } from "react";
import Basket from "./Basket";
import FavoriteCards from "./FavoritCards";
import HomePage from "./HomePage";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { basketSelector, favouriteSelector } from "../../selectors";
import { chengeCategory } from "../../reducers";
import {
	HeaderContainer,
	HeaderWrapp,
	HeaderDescription,
} from "./StyledHeader";

const Header = () => {
	const dispatch = useDispatch();

	const basketCount = useSelector(basketSelector);
	const favoriteCount = useSelector(favouriteSelector);

	useEffect(() => {
		localStorage.setItem("basketCount", JSON.stringify(basketCount));
		localStorage.setItem("favoriteCount", JSON.stringify(favoriteCount));
	}, [basketCount, favoriteCount]);

	return (
		<HeaderContainer>
			<HeaderWrapp>
				<Link to="/" onClick={() => dispatch(chengeCategory(0))}>
					<div className="header__logo">
						<h3 className="logo">Game Store</h3>
						<p className="header__logo-text">магазин компьютерных игр</p>
					</div>
				</Link>
				<HeaderDescription>
					<Link to="/" data-testid="home-link" className="header__homepage">
						<HomePage />
					</Link>
					<Link
						to="api/favorite"
						data-testid="favorite-link"
						className="header__favorite"
					>
						<FavoriteCards />
						<span className="header__cart-number">{favoriteCount.length}</span>
					</Link>
					<Link
						to="api/basket"
						data-testid="basket-link"
						className="header__basket"
					>
						<Basket />
						<span className="header__cart-number">{basketCount.length}</span>
					</Link>
				</HeaderDescription>
			</HeaderWrapp>
		</HeaderContainer>
	);
};

export default Header;
