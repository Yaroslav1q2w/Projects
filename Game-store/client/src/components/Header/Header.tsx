import { useEffect } from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { basketSelector, favouriteSelector } from "../../selectors";
import { chengeCategory, setPageCount } from "../../reducers";
import {
	HeaderContainer,
	HeaderWrapp,
	HeaderDescription,
	HomeIcon,
	FavouriteIcon,
	ShoppingCartIcon,
} from "./StyledHeader";

const Header = () => {
	const dispatch = useDispatch();

	const basketCount = useSelector(basketSelector);
	const favoriteCount = useSelector(favouriteSelector);

	useEffect(() => {
		localStorage.setItem("ShoppingCardCount", JSON.stringify(basketCount));
		localStorage.setItem("FavouriteCount", JSON.stringify(favoriteCount));
	}, [basketCount, favoriteCount]);

	const LogoOnclick = () => {
		dispatch(chengeCategory(0));
		dispatch(setPageCount(1));
	};

	return (
		<HeaderContainer>
			<HeaderWrapp>
				<Link to="/" onClick={LogoOnclick}>
					<div className="header__logo">
						<h3 className="logo">Game Store</h3>
						<p className="header__logo-text">магазин компьютерных игр</p>
					</div>
				</Link>
				<HeaderDescription>
					<Link to="/" data-testid="home-link" className="header__homepage">
						<HomeIcon />
					</Link>
					<Link
						to="api/favorite"
						data-testid="favorite-link"
						className="header__favorite"
					>
						<FavouriteIcon />
						<span className="header__cart-number">{favoriteCount.length}</span>
					</Link>
					<Link
						to="api/basket"
						data-testid="basket-link"
						className="header__basket"
					>
						<ShoppingCartIcon />
						<span className="header__cart-number">{basketCount.length}</span>
					</Link>
				</HeaderDescription>
			</HeaderWrapp>
		</HeaderContainer>
	);
};

export default Header;
