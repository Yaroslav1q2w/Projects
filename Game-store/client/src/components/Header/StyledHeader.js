import styled from "styled-components";
import { AiOutlineHome } from "react-icons/ai";
import { FaStarHalfAlt } from "react-icons/fa";
import { BiCartAlt } from "react-icons/bi";

export const HeaderContainer = styled.header`
	background-color: #282828;
	box-shadow: 0px 8px 10px 0px #282828;
`;

export const HeaderWrapp = styled.header`
	height: 80px;
	max-width: 1180px;
	margin: 0 auto;
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: 0 20px;

	.header__logo {
		.logo {
			font-size: 34px;
			font-weight: bold;
			padding-bottom: 3px;
			text-shadow: 2px 2px 1px #995400;
			color: #ffd700;
			margin: 0;

			@media screen and (max-width: 600px) {
				font-size: 26px;
			}
		}

		&-text {
			font-size: 10px;
			font-weight: 600;
			text-transform: uppercase;
			text-shadow: 1px 1px 1px #5b5b5d;
			color: #c0c0c0;

			@media screen and (max-width: 600px) {
				font-size: 8px;
			}
		}
	}

	@media screen and (max-width: 600px) {
	}
`;

export const HeaderDescription = styled.div`
	display: flex;
	align-items: center;

	.header__favorite {
		padding: 5px;
		font-size: 14px;
		cursor: pointer;

		@media screen and (max-width: 500px) {
			padding: 2px;
		}
	}

	.header__basket {
		padding: 5px;
		cursor: pointer;
		font-size: 14px;

		@media screen and(max-width: 500px) {
			padding: 2px;
		}
	}

	.header__homepage {
		padding-right: 12px;
		cursor: pointer;

		@media screen and (max-width: 500px) {
			padding: 9px;
		}
	}
`;

export const HomeIcon = styled(AiOutlineHome)`
	font-size: 28px;

	@media screen and (max-width: 500px) {
		font-size: 24px;
	}
`;

export const FavouriteIcon = styled(FaStarHalfAlt)`
	font-size: 24px;

	@media screen and (max-width: 500px) {
		font-size: 20px;
	}
`;

export const ShoppingCartIcon = styled(BiCartAlt)`
	font-size: 25px;

	@media screen and (max-width: 500px) {
		font-size: 21px;
	}
`;
