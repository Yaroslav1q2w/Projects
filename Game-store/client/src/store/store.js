import { configureStore } from "@reduxjs/toolkit";

import {
	favoriteReducer,
	todoReducer,
	basketReducer,
	modalReducer,
	categoriesReducer,
} from "../reducers";

const store = configureStore({
	reducer: {
		favorite: favoriteReducer,
		todos: todoReducer,
		basket: basketReducer,
		modal: modalReducer,
		category: categoriesReducer,
	},
});

export default store;
