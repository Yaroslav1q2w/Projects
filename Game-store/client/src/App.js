import React from "react";
import Header from "./components/Header";
import "./App.scss";
import ToDoProvider from "./context";
import RootRouter from "./routers";

const App = () => {
	return (
		<ToDoProvider>
			<div className="wrapper">
				<Header />
				<main className="main__content">
					<RootRouter />
				</main>
			</div>
		</ToDoProvider>
	);
};

export default App;
