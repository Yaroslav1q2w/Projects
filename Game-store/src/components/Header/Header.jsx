import React,{useEffect} from 'react';
import Basket from "./Basket"
import FavoriteCards from "./FavoritCards"
import "./Header.scss"
import HomePage from "./HomePage/HomePage";
import {Link} from "react-router-dom";
import { useSelector } from 'react-redux'
import {basketSelector,favouriteSelector} from "../../selectors";


const Header = () => {
    const basketCount = useSelector(basketSelector)
    const favoriteCount = useSelector(favouriteSelector)

    useEffect(() => {
        localStorage.setItem("basketCount",JSON.stringify(basketCount))
        localStorage.setItem("favoriteCount",JSON.stringify(favoriteCount))
    },[basketCount,favoriteCount])


    return (
        <header className="header">
            <div className="header-wrap">
                <Link to="/">
                    <div className="header__logo">
                        <h3 className="logo">Game Store</h3>
                        <p className="header__logo-text">магазин компьютерных игр</p>
                    </div>
                </Link>
                <div className="header__description">
                    <Link to="/" data-testid="home-link" className="header__homepage">
                        <HomePage />
                    </Link>
                    <Link to="/favorite" data-testid="favorite-link" className="header__favorite">
                        <FavoriteCards />
                        <span className="header__cart-number">{favoriteCount.length}</span>
                    </Link>
                    <Link to="/basket" data-testid="basket-link" className="header__basket">
                        <Basket />
                        <span className="header__cart-number">{basketCount.length}</span>
                    </Link>
                </div>
            </div>
        </header>
    );
};


export default Header;