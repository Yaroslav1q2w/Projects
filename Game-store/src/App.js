import React from 'react';
import Header from "./components/Header";
import "./App.scss"
import ToDoProvider from "./context";
import RootRouter from "./routers";


const App = () => {

    return (
        <ToDoProvider>
            <div className="wrapper" data-testid="home-page">
                <Header />
                <main className="main__content">
                    <RootRouter/>
                </main>
            </div>
        </ToDoProvider>
    );
};

export default App;