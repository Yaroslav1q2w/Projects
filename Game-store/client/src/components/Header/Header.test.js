import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import userEvent from "@testing-library/user-event";
import {Provider} from "react-redux";
import store from "../../store/store"
import Header from "./Header"
import {MemoryRouter} from "react-router-dom";
import App from "../../App";


describe("Header", () => {
    test('Header render' , () => {
        const {container} = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Header  />
                </Provider>
            </MemoryRouter>)

        expect(container).toBeInTheDocument()
    })

    test('Header snapshot' , () => {
        const component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Header  />
                </Provider>
            </MemoryRouter>)

        expect(component).toMatchSnapshot()
    })
})


describe("App", () => {

    test('render router home-page' , () => {
        render(<Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );
        const homeLink = screen.getByTestId('home-link')
        userEvent.click(homeLink)
        expect(screen.getByTestId("home-page")).toBeInTheDocument()
    })

    test('render router favorite-page' , () => {
        render(<Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );
        const favoriteLink = screen.getByTestId('favorite-link')
        userEvent.click(favoriteLink)
        expect(screen.getByTestId("favorite-page")).toBeInTheDocument();
    })

    test('render router basket-page' , () => {
        render(<Provider store={store}>
                <MemoryRouter>
                    <App />
                </MemoryRouter>
            </Provider>
        );
        const basketLink = screen.getByTestId('basket-link')
        userEvent.click(basketLink)
        expect(screen.getByTestId("basket-page")).toBeInTheDocument();
    })
})