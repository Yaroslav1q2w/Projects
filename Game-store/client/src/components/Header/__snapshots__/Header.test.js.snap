// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <a
          class="header__logo"
          href="/"
        >
          Game Store
        </a>
        <div
          class="header__description"
        >
          <a
            class="header__homepage"
            data-testid="home-link"
            href="/"
          >
            <svg
              fill="currentColor"
              font-size="34"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </a>
          <a
            class="header__favorite"
            data-testid="favorite-link"
            href="/favorite"
          >
            <svg
              fill="currentColor"
              font-size="27"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 536 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z"
              />
            </svg>
            <span
              class="header__cart-number"
            >
              0
            </span>
          </a>
          <a
            class="header__basket"
            data-testid="basket-link"
            href="/basket"
          >
            <svg
              fill="currentColor"
              font-size="30"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 4H2v2h2.3l3.28 9a3 3 0 0 0 2.82 2H19v-2h-8.6a1 1 0 0 1-.94-.66L9 13h9.28a2 2 0 0 0 1.92-1.45L22 5.27A1 1 0 0 0 21.27 4 .84.84 0 0 0 21 4zm-2.75 7h-10L6.43 6h13.24z"
              />
              <circle
                cx="10.5"
                cy="19.5"
                r="1.5"
              />
              <circle
                cx="16.5"
                cy="19.5"
                r="1.5"
              />
            </svg>
            <span
              class="header__cart-number"
            >
              0
            </span>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <a
        class="header__logo"
        href="/"
      >
        Game Store
      </a>
      <div
        class="header__description"
      >
        <a
          class="header__homepage"
          data-testid="home-link"
          href="/"
        >
          <svg
            fill="currentColor"
            font-size="34"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
            />
          </svg>
        </a>
        <a
          class="header__favorite"
          data-testid="favorite-link"
          href="/favorite"
        >
          <svg
            fill="currentColor"
            font-size="27"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 536 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z"
            />
          </svg>
          <span
            class="header__cart-number"
          >
            0
          </span>
        </a>
        <a
          class="header__basket"
          data-testid="basket-link"
          href="/basket"
        >
          <svg
            fill="currentColor"
            font-size="30"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 4H2v2h2.3l3.28 9a3 3 0 0 0 2.82 2H19v-2h-8.6a1 1 0 0 1-.94-.66L9 13h9.28a2 2 0 0 0 1.92-1.45L22 5.27A1 1 0 0 0 21.27 4 .84.84 0 0 0 21 4zm-2.75 7h-10L6.43 6h13.24z"
            />
            <circle
              cx="10.5"
              cy="19.5"
              r="1.5"
            />
            <circle
              cx="16.5"
              cy="19.5"
              r="1.5"
            />
          </svg>
          <span
            class="header__cart-number"
          >
            0
          </span>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
